File: JsonUtil.java
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.util.HashMap;
import java.util.Map;

public class JsonUtil {
    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();

    // 美化 JSON
    public static String stringify(Object obj) {
        try {
            return OBJECT_MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(obj);
        } catch (Exception e) {
            return "{}";
        }
    }

    // 解析 JSON
    public static Map<String, Object> parse(String jsonStr) {
        try {
            return OBJECT_MAPPER.readValue(jsonStr, new TypeReference<Map<String, Object>>() {});
        } catch (Exception e) {
            return new HashMap<>();
        }
    }

    // 快速构建 JSON
    public static Map<String, Object> build(Object... keyValues) {
        Map<String, Object> map = new HashMap<>();
        for (int i = 0; i < keyValues.length; i += 2) {
            String key = String.valueOf(keyValues[i]);
            Object value = i + 1 < keyValues.length ? keyValues[i + 1] : null;
            map.put(key, value);
        }
        return map;
    }
}

File: JsonUtil.js
class JsonUtil {
    /**
     * 对象转美化 JSON 字符串
     */
    static stringify(obj) {
        try {
            return JSON.stringify(obj, null, 2);
        } catch (e) {
            return "{}";
        }
    }

    /**
     * 解析 JSON 字符串 → 对象
     */
    static parse(jsonStr) {
        try {
            return JSON.parse(jsonStr);
        } catch (e) {
            return {};
        }
    }

    /**
     * 快速构建 JSON（直接传键值对）
     */
    static build(...args) {
        const obj = {};
        for (let i = 0; i < args.length; i += 2) {
            const key = args[i];
            const value = args[i + 1];
            if (key) obj[key] = value;
        }
        return obj;
    }
}

module.exports = JsonUtil;
